userspace/libsinsp/examples/test.cpp

```diff
@@ -334,151 +347,81 @@ sinsp_evt* get_event(sinsp& inspector, std::function<void(const std::string&)> h
 	return nullptr;
 }
 
-void plaintext_dump(sinsp& inspector)
+void formatted_dump(sinsp&, sinsp_evt* ev)
 {
-
-	sinsp_evt* ev = get_event(inspector, [](const std::string& error_msg)
-				  { cout << "[ERROR] " << error_msg << endl; });
-
-	if(ev == nullptr)
+	std::string output;
+	if(ev->get_category() == EC_PROCESS)
 	{
-		return;
+		process_formatter->tostring(ev, output);
 	}
-
-	sinsp_threadinfo* thread = ev->get_thread_info();
-	if(thread)
+	else
 	{
-		string cmdline;
-		sinsp_threadinfo::populate_cmdline(cmdline, thread);
-
-		if(g_all_threads || thread->is_main_thread())
-		{
-			string date_time;
-			sinsp_utils::ts_to_iso_8601(ev->get_ts(), &date_time);
+		default_formatter->tostring(ev, output);
+	}
 
-			bool is_host_proc = thread->m_container_id.empty();
-			cout << "[" << date_time << "]:["
-			     << (is_host_proc ? "HOST" : thread->m_container_id) << "]:";
+	cout << output << std::endl;
+}
 
-			cout << "[CAT=";
+static void hexdump(const char* buf, size_t len)
+{
+	bool in_ascii = false;
 
-			if(ev->get_category() == EC_PROCESS)
-			{
-				cout << "PROCESS]:";
-			}
-			else if(ev->get_category() == EC_NET)
+	putc('[', stdout);
+	for(size_t i = 0; i < len; ++i)
+	{
+		if(isprint(buf[i]))
+		{
+			if(!in_ascii)
 			{
-				cout << get_event_category_name(ev->get_category()) << "]:";
-				sinsp_fdinfo_t* fd_info = ev->get_fd_info();
-
-				// event subcategory should contain SC_NET if ipv4/ipv6
-				if(nullptr != fd_info && (fd_info->get_l4proto() != SCAP_L4_UNKNOWN && fd_info->get_l4proto() != SCAP_L4_NA))
+				in_ascii = true;
+				if(i > 0)
 				{
-					cout << "[" << fd_info->tostring() << "]:";
+					putc(' ', stdout);
 				}
+				putc('"', stdout);
 			}
-			else if(ev->get_category() == EC_IO_READ || ev->get_category() == EC_IO_WRITE)
+			putc(buf[i], stdout);
+		}
+		else
+		{
+			if(in_ascii)
 			{
-				cout << get_event_category_name(ev->get_category()) << "]:";
-
-				sinsp_fdinfo_t* fd_info = ev->get_fd_info();
-				if(nullptr != fd_info && (fd_info->get_l4proto() != SCAP_L4_UNKNOWN && fd_info->get_l4proto() != SCAP_L4_NA))
-				{
-					cout << "[" << fd_info->tostring() << "]:";
-				}
+				in_ascii = false;
+				fputs("\" ", stdout);
 			}
-			else
+			else if(i > 0)
 			{
-				cout << get_event_category_name(ev->get_category()) << "]:";
+				putc(' ', stdout);
 			}
-
-			sinsp_threadinfo* p_thr = thread->get_parent_thread();
-			int64_t parent_pid = -1;
-			if(nullptr != p_thr)
-			{
-				parent_pid = p_thr->m_pid;
-			}
-
-			cout << "[PPID=" << parent_pid << "]:"
-			     << "[PID=" << thread->m_pid << "]:"
-			     << "[TYPE=" << get_event_type_name(inspector, ev) << "]:"
-			     << "[EXE=" << thread->get_exepath() << "]:"
-			     << "[CMD=" << cmdline << "]"
-			     << endl;
+			printf("%02x", buf[i] & 0xff);
```

Comment:
Let me check that. I'm pretty sure the value gets sign-extended (since char is generally signed) so bytes over 0x7f would render as e.g. ffffff80. The `& 0xff` looks redundant (it gets applied before the implicit cast to int) but (AIUI) the type of 0xff is int, which means buf[i] gets promoted to an int before getting ANDed, so the value passed to printf() is an int with the zero bits cleared.